connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestServer.java

```diff
@@ -306,6 +300,18 @@ public void initializeResources(Herder herder) {
         log.info("REST resources initialized; server is started and ready to handle requests");
     }
 
+    protected abstract Collection<ConnectResource> regularResources();
+
+    protected abstract Collection<ConnectResource> adminResources();
+
+    protected void configureRegularResources(ResourceConfig resourceConfig) {
```

Comment:
I find the current name confusing, is this more or less clear?
```suggestion
    protected void registerRegularResources(ResourceConfig resourceConfig) {
```

Reply:
IMO `register` is misleading since it implies that it's the class's responsibility to ensure that the resources returned from `RestServer::regularResources` are now registered with the given `resourceConfig`, when in fact, that's already taken care of automatically and this method is (for now) just a convenient hook to initialize REST extensions.

Still happy to rename to something clearer if you have ideas, but as a preemptive measure I've also Javadoc'd this methods and others like it to hopefully make things a bit clearer.